source/Ship.h

```diff
@@ -339,7 +343,10 @@ class Ship : public Body, public std::enable_shared_from_this<Ship> {
 	// DamageDealt from that weapon. The return value is a ShipEvent type,
 	// which may be a combination of PROVOKED, DISABLED, and DESTROYED.
 	// Create any target effects as sparks.
-	int TakeDamage(std::vector<Visual> &visuals, const DamageDealt &damage, const Government *sourceGovernment);
+	int TakeDamage(std::vector<Visual> &visuals, const DamageDealt &damage, const Government *sourceGovernment, Point intersection, Point hitVelocity, Angle hitAngle);
+	// Apply the actual damage to the ship, in the case of weather, this
+	// is called directly and TakeDamage is skipped
```

Comment:
Why wouldn't hazards generate hit effects?

Reply:
Everywhere seems fine to me lol. If the hit effect is independent from the weapon then logically (at least imo) it should also apply to hazards that do actual damage. Having a separate hazard hit effect doesn't really makes sense to me.

Reply:
I think it makes sense for hazards to have a custom hit effect, but it doesn't have to be a part of this PR, it could be as a follow up.
A meteor shower could for instance make small explosions happen when it hits your ship, but it does mean we have to wonder where they happen and they can't be everywhere else it'll look so full of them, instead it should have a chance of appearing on different points, depending on the strength of the hazard, probably?

Reply:
Right but that shouldn't be a hit effect on the ship itself.

Reply:
You mean shields reacting to that damage?
Both would probably be at the same location, and using the same logic to spawn them there; whether it is based on the amount of damage taken, the strength of the hazards etc, and we may not give it justice here, though I don't personally mind big PRs.

I guess that for now we could just have the shields reacting randomly to the hazards, say simply 10% chance of each part of the ship to react and see where that goes? It would probably have quite a big performance impact, however.

Reply:
For weather damage, I could do some random rolls to see if it does any effects (and this can be changed later) as it doesn't make sense for the effects to just pile on top of each other endlessly. Similarly for blast weapons I could spawn a random number of effects to avoid having them plaster the entire ship but rather simulate debris smashing into the shields etc..